js/web/components/Hub.js

```diff
@@ -0,0 +1,272 @@
+const HubHeader = dynamic(() => import('./HubHeader'))
+const TabHeader = dynamic(() => import('./TabHeader'))
+const ReusableTable = dynamic(() => import('./ReusableTable'))
+
+const HubComponent = ({ hubPubkey }) => {
+  const {
+    getHub,
+    hubState,
+    filterHubContentForHub,
+    filterHubCollaboratorsForHub,
+    hubContentState,
+  } = useContext(Hub.Context)
+
+  const { releaseState } = useContext(Release.Context)
+
+  const [hubReleases, setHubReleases] = useState(undefined)
+  const [releaseData, setReleaseData] = useState(undefined)
+  const [collaboratorsData, setCollaboratorsData] = useState(undefined)
+  const [activeView, setActiveView] = useState(undefined)
+  const [fetchedHubInfo, setFetchedHubInfo] = useState(false)
+  const [fetchedReleases, setFetchedReleases] = useState(false)
+  const [fetchedCollaborators, setFetchedCollaborators] = useState(false)
+  const [invalidHubPubkey, setInvalidHubPubkey] = useState(false)
+  const [views, setViews] = useState([
+    { name: 'releases', playlist: undefined, visible: true },
+    { name: 'collaborators', playlist: undefined, visible: true },
+  ])
+  const hubData = useMemo(() => hubState[hubPubkey], [hubState, hubPubkey])
+
+  useEffect(() => {
+    if (!hubPubkey) {
+      setFetchedHubInfo(false)
+    }
+    getHub(hubPubkey)
+    if (hubPubkey) {
+      setFetchedHubInfo(true)
+    }
+    if (fetchedHubInfo && !hubPubkey) {
+      setInvalidHubPubkey(true)
+    }
+  }, [hubPubkey])
+
+  useEffect(() => {
+    const [releases] = filterHubContentForHub(hubPubkey)
+    const collaborators = filterHubCollaboratorsForHub(hubPubkey)
+    setHubReleases(releases)
+    setCollaboratorsData(collaborators)
+    let updatedView = views.slice()
+    let viewIndex
+    if (releases.length > 0) {
+      setActiveView(0)
+      viewIndex = updatedView.findIndex((view) => view.name === 'releases')
+      updatedView[viewIndex].visible = true
+      updatedView[viewIndex].playlist = releases
+      setFetchedReleases(true)
+    }
+    if (releases.length === 0 && collaborators.length > 0) {
+      setActiveView(1)
+      setFetchedReleases(false)
+    }
+    if (collaborators) {
+      setFetchedCollaborators(true)
+    }
+    setViews(updatedView)
+  }, [hubContentState])
+
+  useEffect(() => {
+    let updatedView = views.slice()
+    let viewIndex
+    const data = hubReleases?.map((hubRelease) => {
+      const releaseMetadata = releaseState.metadata[hubRelease.release]
+      releaseMetadata.releasePubkey = hubRelease.release
+      return releaseMetadata
+    })
+    setReleaseData(data)
+    viewIndex = updatedView.findIndex((view) => view.name === 'releases')
+    updatedView[viewIndex].playlist = releaseData
+  }, [releaseState, hubReleases, views])
+
+  const viewHandler = (event) => {
+    const index = parseInt(event.target.id)
+    setActiveView(index)
+  }
+  const noHubs = releaseData?.length === 0 && collaboratorsData?.length === 0;
```

Comment:
should be in a useEffect with the resource fetch lsiteners in the dependency array